sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -173,7 +173,7 @@ class RegexParser(pattern: String) {
   }
 
   private def parseCharacterClass(): RegexCharacterClass = {
-    val supportedMetaCharacters = "\\^-]+"
+    val supportedMetaCharacters = "\\^-[]+"
```

Comment:
I think this is related to what has to be escaped inside a character class. I don't think you need to escape `[` in a character class, ie `[[]` might be valid actually.

Reply:
Wouldn't that be a nested character class then, which needs to be closed with a `]`? Looks like it doesn't work:
```scala
scala> "[test[abc]".r
java.util.regex.PatternSyntaxException: Unclosed character class near index 9
[test[abc]
         ^
  at java.util.regex.Pattern.error(Pattern.java:1969)
```

Reply:
BTW, I added this in this PR because I ran into an issue where I was re-parsing an already transpiled string to get it back to AST form and it would fail here.

Reply:
I'm pretty sure that unescaped `[` characters are supported in character classes, but it does depend on the context.

Reply:
This is supposed to be a list of the meta characters that _can_ be escaped in a character class, so `[` should belong here.

Technically there are still other escapable characters (eg. something like `[\#]` is valid) but it isn't documented anywhere. In these cases the escape doesn't do anything so I don't think anyone would actually use this.